chalk-ir/src/zip.rs

```diff
@@ -397,16 +398,18 @@ impl<I: Interner> Zip<I> for GoalData<I> {
     }
 }
 
+// FIXME(areredify) support consts
+
 // I'm too lazy to make `enum_zip` support type parameters.
-impl<T: Zip<I>, L: Zip<I>, I: Interner> Zip<I> for ParameterKind<T, L> {
+impl<T: Zip<I>, L: Zip<I>, C: Debug, I: Interner> Zip<I> for ParameterKind<T, L, C> {
     fn zip_with<'i, Z: Zipper<'i, I>>(zipper: &mut Z, a: &Self, b: &Self) -> Fallible<()>
     where
         I: 'i,
     {
         match (a, b) {
             (ParameterKind::Ty(a), ParameterKind::Ty(b)) => Zip::zip_with(zipper, a, b),
             (ParameterKind::Lifetime(a), ParameterKind::Lifetime(b)) => Zip::zip_with(zipper, a, b),
-            (ParameterKind::Ty(_), _) | (ParameterKind::Lifetime(_), _) => {
+            _ => {
```

Comment:
This is error-prone -- if we add a new variant, we get no error... I'd prefer to see

```rust
(ParameterKind::Ty(_), _) |
(ParameterKind::Const(_), _) |
(ParameterKind::Lifetime(_), _) => {
}
```

Reply:
This is an outdated piece of code